### PR TITLE
[Issue #7583][build] - Changed pom to be consistent with version property placeholders

### LINKED ISSUE #7583: Root pom is inconsistent with dependency versions as property variables
Most of the version declarations in the root pom are ${} placeholders, a few are not.
An example of this is

      
      <dependency>
        <groupId>com.google.guava</groupId>
        <artifactId>guava</artifactId>
        <version>${guava.version}</version>
      </dependency>

     <dependency>
        <groupId>com.beust</groupId>
        <artifactId>jcommander</artifactId>
        **<version>1.48</version>**
      </dependency>

      <dependency>
        <groupId>org.apache.commons</groupId>
        <artifactId>commons-lang3</artifactId>
        **<version>3.6</version>**
      </dependency>

      <dependency>
        <groupId>org.apache.commons</groupId>
        <artifactId>commons-compress</artifactId>
        <version>${commons-compress.version}</version>
      </dependency>

      <dependency>
        <groupId>commons-configuration</groupId>
        <artifactId>commons-configuration</artifactId>
        **<version>1.10</version>**
      </dependency>

      <dependency>
        <groupId>commons-io</groupId>
        <artifactId>commons-io</artifactId>
        **<version>2.5</version>**
      </dependency>

Update the root pom to follow maven recommended conventions.

### PR DESCRIPTION
**Pull request by vcottagiri**
Fixes #7583

### Motivation
- Root pom is inconsistent with dependency versions as property variables

### Modifications

- Moved version declarations for dependency managed libs to property variables
- Moved version declarations for maven plugins to property variables
- Removed io.netty:netty:3.10.6.Final as its not used directly in any of the pulsar sub-projects
- Removed duplicate definition of maven-enforcer-plugin:1.0.1

### Verifying this change

- [X] Make sure that the change passes the CI checks.

### CODE REVIEW

**Comment by vcottagiri** [comment]
Looking into the build failures

File: pom.xml [hunk 1]
```diff
@@ -662,12 +690,6 @@ flexible messaging model and an intuitive client API.</description>
         <version>${jersey.version}</version>
       </dependency>
 
-      <dependency>
-        <groupId>net.java.dev.jna</groupId>
```
Comment (merlimat) [inline pom.xml:666]: This is needed to pin the dependency version

Comment (vcottagiri) [reply pom.xml:666]: ok, will revert

**Comment by vcottagiri** [comment]
@merlimat - The jna dependency comes from bookkeeper-server dependency, Pulsar does not directly use it. bookkeeper-server uses jna-3.2.7, whats the reason we need to upgrade to jna-4.2.0?

Also, noticed that pulsar-presto-distribution brings in jna-5.3.1 transitively


> [INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ pulsar-presto-distribution ---  
> [INFO] org.apache.pulsar:pulsar-presto-distribution:jar:2.7.0-SNAPSHOT  
> [INFO] \\- io.prestosql:presto-cli:jar:332:compile  
> [INFO]    \\- org.jline:jline-terminal-jna:jar:3.12.1:runtime  
> [INFO]      \\- net.java.dev.jna:jna:jar:5.3.1:runtime

Please do confirm if this is the intended behavior. I have attached the output of the mvn dependency:tree cmd for jna
[jna.txt](https://github.com/apache/pulsar/files/4950713/jna.txt)

**Comment by merlimat** [comment]
@vcottagiri We pinned to 4.2.0 long time back in #560 for licensing reason, so it should really be set to this.

**Pull request merged**